torch/csrc/jit/passes/symbolic_shape_analysis.h

```diff
@@ -47,5 +48,63 @@ PropagateShapesAndBuildLargeShapeComputeGraph(
 TORCH_API bool setSymbolicShapeAnalysisTestMode(bool value);
 TORCH_API bool symbolicShapeAnalysisTestModeEnabled();
 
+using SSAInput = c10::variant<IValue, c10::SymbolicShape>;
+TORCH_API c10::optional<std::vector<c10::SymbolicShape>>
+calculateSymbolicShapesOnOp(
+    const FunctionSchema* schema,
+    const std::vector<SSAInput>& inputs);
+
+struct TORCH_API CanonicalizedSymbolicShape {
+  CanonicalizedSymbolicShape(c10::SymbolicShape orig_shape) {
```

Comment:
This is only needed for caching right ? we're pretty far from caching so I think it would make sense to delay this PR. Also I think you want to Canonicalize with respect to all arguments of an a node, not a single value. E.g. right now if you try Canonicalize on `Size(SS(-3)), Size(SS(-4))` youd get `(Size(SS(-1), SS(-1))` when in fact they are not equal

Reply:
No, I am using this as a utility item to test the rest of this diff (Have you not been reading my responses?).
You do raise a good point about the fact that we need to conanicalize based on all of the shapes in the node.